tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,237 @@
 
     assert result["type"] == "abort"
 
 
-async def test_check_zigpy_connection():
-    """Test config flow validator."""
+async def test_probe_radios(hass):
+    """Test detect radios."""
+    app_ctrl_cls = MagicMock()
+    app_ctrl_cls.SCHEMA_DEVICE = zigpy.config.SCHEMA_DEVICE
+    app_ctrl_cls.probe = AsyncMock(side_effect=(True, False))
 
-    mock_radio = tests.async_mock.MagicMock()
-    mock_radio.connect = tests.async_mock.AsyncMock()
-    radio_cls = tests.async_mock.MagicMock(return_value=mock_radio)
+    with patch.dict(config_flow.RADIO_TYPES, {"ezsp": {CONTROLLER: app_ctrl_cls}}):
+        res = await config_flow.detect_radios("/dev/null")
+        assert app_ctrl_cls.probe.await_count == 1
+        assert res[CONF_RADIO_TYPE] == "ezsp"
+        assert zigpy.config.CONF_DEVICE in res
+        assert (
+            res[zigpy.config.CONF_DEVICE][zigpy.config.CONF_DEVICE_PATH] == "/dev/null"
+        )
 
-    bad_radio = tests.async_mock.MagicMock()
-    bad_radio.connect = tests.async_mock.AsyncMock(side_effect=Exception)
-    bad_radio_cls = tests.async_mock.MagicMock(return_value=bad_radio)
+        res = await config_flow.detect_radios("/dev/null")
+        assert res is None
 
-    mock_ctrl = tests.async_mock.MagicMock()
-    mock_ctrl.startup = tests.async_mock.AsyncMock()
-    mock_ctrl.shutdown = tests.async_mock.AsyncMock()
-    ctrl_cls = tests.async_mock.MagicMock(return_value=mock_ctrl)
-    new_radios = {
-        mock.sentinel.radio: {ZHA_GW_RADIO: radio_cls, CONTROLLER: ctrl_cls},
-        mock.sentinel.bad_radio: {ZHA_GW_RADIO: bad_radio_cls, CONTROLLER: ctrl_cls},
-    }
 
-    with mock.patch.dict(
-        homeassistant.components.zha.core.registries.RADIO_TYPES, new_radios, clear=True
-    ):
-        assert not await config_flow.check_zigpy_connection(
-            mock.sentinel.usb_path, mock.sentinel.unk_radio, mock.sentinel.zigbee_db
+async def test_user_port_config_fail(hass):
+    """Test port config flow."""
+    app_ctrl_cls = MagicMock()
+    app_ctrl_cls.SCHEMA_DEVICE = zigpy.config.SCHEMA_DEVICE
+    app_ctrl_cls.probe = AsyncMock(return_value=False)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: "pick_radio"}, data={CONF_RADIO_TYPE: "ezsp"}
+    )
+
+    with patch.dict(config_flow.RADIO_TYPES, {"ezsp": {CONTROLLER: app_ctrl_cls}}):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={zigpy.config.CONF_DEVICE_PATH: "/dev/ttyUSB33"},
         )
-        assert mock_radio.connect.call_count == 0
-        assert bad_radio.connect.call_count == 0
-        assert mock_ctrl.startup.call_count == 0
-        assert mock_ctrl.shutdown.call_count == 0
-
-        # unsuccessful radio connect
-        assert not await config_flow.check_zigpy_connection(
-            mock.sentinel.usb_path, mock.sentinel.bad_radio, mock.sentinel.zigbee_db
+        assert result["type"] == RESULT_TYPE_FORM
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)